pyproject.toml

```diff
@@ -29,3 +62,111 @@ ignore_missing_imports = true
 disallow_untyped_calls = true
 disallow_untyped_defs = true
 disallow_incomplete_defs = true
+
+[tool.poetry.dependencies]
+python = ">=3.7,<3.11"
+pydicom = "^2.3.0"
+typing-extensions = "*"
+setuptools = "*"
+
+# ----- #
+# Apps  #
+# ----- #
+sqlalchemy = { version =">=2.0", optional = true } # groups ["all", "tests", "dev" ]
+
+# ------------- #
+# Documentation #
+# ------------- #
+sphinx = { version = "5.3.0", optional = true } # groups = ["all", "docs"]
+sphinx-argparse = { version = "*", optional = true } # groups = ["all", "docs"]
+sphinx_copybutton = { version = "*", optional = true } # groups = [ "all", "docs"]
+sphinx_rtd_theme = { version = "*", optional = true } # groups = [ "all", "docs"]
+# sphinx-ext-napoleon = { version = "*", optional = true }  # groups = ["all", "docs"]
+# sphinx-book-theme = { version = "*", optional = true }   # groups = ["all", "docs", "doctests"]
+# jupyter-book = { version = ">=0.12.0", optional = true } # groups = ["all", "docs"]
+# networkx = { version = "*", optional = true }            # groups = ["all", "docs"]
+
+# ----- #
+# Tests #
+# ----- #
+pytest = { version = "*", optional = true }                 # groups = ["all", "tests"]
+pytest-sugar = { version = "*", optional = true }           # groups = ["all", "tests"]
+# hypothesis = { version = "<6", optional = true }          # groups = ["all", "tests"]
+# astroid = { version = "*", optional = true }              # groups = ["all", "tests"]
+# psutil = { version = "*", optional = true }               # groups = ["all", "tests"]
+pylint = { version = "*", optional = true }                 # groups = ["all", "tests"]
+pytest-rerunfailures = { version = "*", optional = true }   # groups = ["all", "tests"]
+pytest-cov = { version = "*", optional = true }             # groups = ["all", "tests"]
+coverage = { version = "*", optional = true }               # groups = ["all", "tests"]
+
+# ---------------- #
+# Development Only #
+# ---------------- #
+pre-commit = { version = "*", optional = true }      # groups = ["dev", "all"]
```

Comment:
`pre-commit` would also make sense to setup, but we we don't use it, right?
Maybe I don't understand the purpose of all these optional packages - are they there to facilitate local checks?

Reply:
I agree that we should use precommit hooks.

Reply:
> Maybe I don't understand the purpose of all these optional packages - are they there to facilitate local checks?

Yes, intended for local checking.
A next step would be to have most of the CI/CD checks defined so that they could be run locally using poetry, e.g.
`poetry run tests`
And instead of explicit build for docs using Make, a similar thing could be done via poetry, or at least the html smoke test/doc generation.
@SimonBiggs knows more.